Source/GoBot/ViewDatabase.swift

```diff
@@ -1594,6 +1651,26 @@ class ViewDatabase {
         }
     }
     
+    func posts(matching text: String) throws -> [KeyValue] {
+        guard let connection = self.openDB else {
+            throw ViewDatabaseError.notOpen
+        }
+        
+        // probably need to escape some characters here
+        var messages = [KeyValue]()
+        let query = try connection.prepare(
+            postSearch
+                .filter(colText.match(text))
+        )
+        for row in query {
+            let messageID = row[colMessageRef]
+            let message = try post(with: messageID)
```

Comment:
I wonder what's the performance of this, because we are executing a query (for the search), and then N queries (one for each result), and the mapQueryToKeyValue func inside run more queries as well. I get it that they should be fast queries so we might leave it as it is and optimize this later when we refactor the discover algorithm into a single sql query as the others.

Reply:
Yeah this is definitely a very naive way to do it. Combining this into one query would be a great optimization in the future, but I'd like to leave it this way for now for the sake of time.